scripts/devops_tasks/test_run_samples.py

```diff
@@ -87,45 +107,117 @@
+    Don't want to break anyone that's using the original code.
+    """
+    try:
+        logging.info(
+            "Command Array: {0}, Target Working Directory: {1}".format(
+                command_array, working_directory
+            )
+        )
+        check_call(command_array, cwd=working_directory, timeout=timeout)
+    except CalledProcessError as err:
+        if err.returncode not in acceptable_return_codes:
+            logging.error(err)  # , file = sys.stderr
+            if always_exit:
+                exit(1)
+            else:
+                return err
+    except TimeoutExpired as err:
+        if pass_if_timeout:
+            logging.info(
+                "Sample timed out successfully"
+            )
+        else:
+            logging.info(
+                "Fail: Sample timed out"
+            )
+            return err
+
+
+def execute_sample(sample, samples_errors, timed):
+    if isinstance(sample, tuple):
+        sample, timeout, pass_if_timeout = sample
+
+    if sys.version_info < (3, 5) and sample.endswith("_async.py"):
+        return
+
+    logging.info(
+        "Testing {}".format(sample)
+    )
+    command_array = [sys.executable, sample]
+
+    if not timed:
+        errors = run_check_call(command_array, root_dir)
+    else:
+        errors = run_check_call_with_timeout(
+            command_array, root_dir, timeout, pass_if_timeout
+        )
+
+    sample_name = os.path.basename(sample)
+    if errors:
+        samples_errors.append(sample_name)
+        logging.info(
+            "ERROR: {}".format(sample_name)
+        )
+    else:
+        logging.info(
+            "SUCCESS: {}.".format(sample_name)
+        )
+
+
 def run_samples(targeted_package):
     logging.info("running samples for {}".format(targeted_package))
 
     samples_errors = []
     sample_paths = []
+    timed_sample_paths = []
+
     samples_dir_path = os.path.abspath(os.path.join(targeted_package, "samples"))
     package_name = os.path.basename(targeted_package)
+    samples_need_timeout = TIMEOUT_SAMPLES.get(package_name, {})
+
+    # install extra dependencies for samples if needed
+    try:
+        with open(samples_dir_path + "/sample_dev_requirements.txt") as sample_dev_reqs:
+            for dep in sample_dev_reqs.readlines():
+                check_call([sys.executable, '-m', 'pip', 'install', dep])
+    except FileNotFoundError:
+        pass
 
     for path, subdirs, files in os.walk(samples_dir_path):
         for name in files:
-            if fnmatch(name, "*.py") and name not in IGNORED_SAMPLES.get(package_name, []):
+            if fnmatch(name, "*.py") and name in samples_need_timeout:
+                timeout = samples_need_timeout[name]
+                # timeout, pass_if_timeout is True by default if nothing passed in
+                if isinstance(timeout, tuple):
+                    timeout, pass_if_timeout = timeout
+                else:
+                    pass_if_timeout = True
```

Comment:
@kristapratico , added the default here in order to not change the flow of the current code too much. Made both `timeout` and `pass_if_timeout` required params in `run_check_call_with_timeout` since it seems that there would be no reason both of those wouldn't be passed in. Does that seem reasonable?

Reply:
Sounds good to me!

Reply:
Ah just realized I can't approve my own PR. I'm ready to merge when you are.

Reply:
sounds good!